src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/cas1/seed/SeedFromXLSXApprovedPremisesRoomsTest.kt

```diff
@@ -0,0 +1,436 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.integration.cas1.seed
+
+import org.assertj.core.api.Assertions.assertThat
+import org.jetbrains.kotlinx.dataframe.api.dataFrameOf
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+import org.springframework.beans.factory.annotation.Autowired
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.api.model.SeedFromExcelFileType
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.integration.seed.SeedTestBase
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.seed.cas1.SiteSurvey
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.seed.cas1.SiteSurveyImportException
+import java.util.UUID
+
+@TestInstance(TestInstance.Lifecycle.PER_METHOD)
+class SeedFromXLSXApprovedPremisesRoomsTest : SeedTestBase() {
+
+  @Autowired
+  lateinit var siteSurvey: SiteSurvey
+
+  fun MutableList<String>.addCharacteristics(numberOfRooms: Int = 1, activeCharacteristics: Map<String, List<Int>> = emptyMap()) {
+    siteSurvey.questionToCharacterEntityMapping.keys.forEach { question ->
+      this.add(question)
+      val answers = MutableList(numberOfRooms) { "No" }
+      activeCharacteristics[question]?.forEach {
+        answers[it] = "Yes"
+      }
+      this.addAll(answers)
+    }
+  }
+
+  @Test
+  fun `Creating a new room and a new bed with a characteristic succeeds`() {
+    val localAuthorityArea = localAuthorityEntityFactory.produceAndPersist()
+    val probationRegion = probationRegionEntityFactory.produceAndPersist()
+    val premises = approvedPremisesEntityFactory.produceAndPersist {
+      withLocalAuthorityArea(localAuthorityArea)
+      withProbationRegion(probationRegion)
+    }
+    val premisesId = premises.id
+
+    val header = listOf("Unique Reference Number for Bed", "SWABI01NEW")
+    val rows = mutableListOf(
+      "Room Number / Name",
+      "1",
+      "Bed Number (in this room i.e if this is a single room insert 1.  If this is a shared room separate entries will need to be made for bed 1 and bed 2)",
+      "1",
+    )
+    rows.addCharacteristics(1, mapOf("Is this room located on the ground floor?" to listOf(0)))
+
+    val dataFrame = dataFrameOf(header, rows)
+
+    withXlsx("example", "Sheet3", dataFrame)
+
+    seedService.seedExcelData(
+      SeedFromExcelFileType.approvedPremisesRoomFromExcel,
+      premisesId,
+      "example.xlsx",
+    )
+
+    val newRoom = roomRepository.findByCode("SWABI01NEW")
+    assertThat(newRoom!!.characteristics).anyMatch {
+      it.name == "Is this room located on the ground floor?" &&
+        it.propertyName == "isGroundFloor"
+    }
+
+    val newBed = bedRepository.findByCodeAndRoomId("SWABI01NEW - 1", newRoom.id)
+    assertThat(newBed!!.name).isEqualTo("SWABI01NEW")
+    assertThat(
+      newBed.room.id == newRoom.id &&
+        newBed.room.code == newRoom.code,
```

Comment:
I think when we assert on `newBed.room.code` we can use a String literal? Same goes for anywhere else doing a similar assertions

Reply:
Changed to use string literals where applicable.